### PR TITLE
Implement --to-ports flag to specify explicit masquerading to-ports

### PR DESCRIPTION
**Pull request by jingyuanliang**
/assign @MrHohn

### CODE REVIEW

File: cmd/ip-masq-agent/ip-masq-agent.go [hunk 1]
```diff
@@ -46,8 +48,14 @@ const (
...
 	// name of nat chain for iptables masquerade rules
-	masqChain                         utiliptables.Chain
-	masqChainFlag                     = flag.String("masq-chain", "IP-MASQ-AGENT", `Name of nat chain for iptables masquerade rules.`)
+	masqChain     utiliptables.Chain
+	masqChainFlag = flag.String("masq-chain", "IP-MASQ-AGENT", `Name of nat chain for iptables masquerade rules.`)
+
+	// create MASQUERADE iptables rules with --to-ports flag
+	toPorts          interval.Intervals
+	toPortsProtocols = []string{"tcp", "udp", "sctp"}
+	toPortsFlag      = flag.String("to-ports", "", fmt.Sprintf(`Masquerade to specified ports only, example: "1024-29999,32768-65535"; applicable to %s protocols.`, strings.Join(toPortsProtocols, ", ")))
```
Comment (MrHohn) [inline cmd/ip-masq-agent/ip-masq-agent.go:57]: I know we have not been doing a good job on keeping the doc updated - https://github.com/kubernetes-sigs/ip-masq-agent/blob/master/README.md#agent-flags seems outdated now.

**Review by MrHohn — COMMENTED** [review]
Thanks Jingyuan, LGTM with minor comments.

File: cmd/ip-masq-agent/ip-masq-agent.go [hunk 2]
```diff
@@ -139,6 +147,14 @@ func main() {
 
 	masqChain = utiliptables.Chain(*masqChainFlag)
 
+	if *toPortsFlag != "" {
+		tp, err := interval.ParseIntervals(*toPortsFlag)
+		if err != nil {
+			klog.Exitf("Invalid --to-ports flag: %s", *toPortsFlag)
```
[the remark this answers is not in the document]
Comment (jingyuanliang) [reply cmd/ip-masq-agent/ip-masq-agent.go]: done.

File: cmd/ip-masq-agent/ip-masq-agent.go (hunk 1 above)
Comment (jingyuanliang) [reply cmd/ip-masq-agent/ip-masq-agent.go:57]: Let's move doc updates to a separate PR? I see some other changes needed (e.g. random-fully) and don't want to mix it in here, or causing merge conflicts.

**Review by MrHohn — APPROVED** [review]
/lgtm

**Pull request merged**